### PR TITLE
DAOS-10956 test: fail the test for ENOMEM

### PR DESCRIPTION
**Pull request by gnailzenh**
Addresses CID: 21986

fail the test for ENOMEM

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

### CODE REVIEW

**Comment by kprantis** [comment]
I think this PR needs to merge in master to be able to pass.

**Review by wangdi1 — APPROVED** [review]

File: src/common/tests/btree.c [hunk 1]
```diff
@@ -830,8 +830,10 @@ ik_btr_drain(void **state)
 	int		 i;
 
 	D_ALLOC_ARRAY(arr, drain_keys);
-	if (arr == NULL)
+	if (arr == NULL) {
 		fail_msg("Array allocation failed");
+		fail();
```
Comment (jolivier23) [inline src/common/tests/btree.c:835]: doesn't fail_msg abort the test?

Comment (gnailzenh) [reply src/common/tests/btree.c]: indeed, I guess we need to call return explicitly to make coverity happy

File: src/common/tests/btree.c [hunk 2]
```diff
@@ -704,8 +704,10 @@ ik_btr_batch_oper(void **state)
 	}
 
 	D_ALLOC_ARRAY(arr, key_nr);
-	if (arr == NULL)
+	if (arr == NULL) {
 		fail_msg("Array allocation failed");
```
Comment (wangshilong) [inline src/common/tests/btree.c:708]: I did not find definition of fail_msg() will this function quit program as well? or it just output message.

**Review by wangshilong — APPROVED** [review]

File: src/common/tests/btree.c (hunk 2 above)
Comment (NiuYawei) [reply src/common/tests/btree.c:708]: According to cmocka manual, fail_msg() forces the test to fail immediately and quit, printing the reason, so I think it's not necessary to call 'return' herel.

Comment (wangshilong) [reply src/common/tests/btree.c:708]: So coverity is not smart enough, we could mark coverity as negative or return here to make it happy.

**Review by NiuYawei — APPROVED** [review]

**Comment by wangshilong** [comment]
@daos-stack/daos-gatekeeper the only failure is dfs_par test which should not be related this PR

**Pull request merged**